pkg/reconciler/ingress/ingress.go

```diff
@@ -192,12 +205,18 @@ func markIngressReady(ingress *v1alpha1.Ingress) {
 	ingress.Status.MarkNetworkConfigured()
 }
 
-// TODO(bvennam) we need to pull this from the upstream ingress that is create "-new"
 func domainForLocalGateway(ingressName string, isPrivate bool) string {
-	if isPrivate {
-		return privateLBDomain
+	if LBDomain, ok := loadBalancers[strings.Split(ingressName, ".")[0]]; ok {
+		if isPrivate {
+			return LBDomain.Private
+		}
```

Comment:
this if statement can be rewritten in a easier fashion.

Reply:
It’s confusing right now since checking for isPrivate twice

Reply:
Split into helper funcs if needed

Reply:
Split each case (known good from available, or default if none known good are found) into a helper function.